src/pokemon.c

```diff
@@ -6941,3 +6942,198 @@ void UpdateDaysPassedSinceFormChange(u16 days)
         }
     }
 }
+
+u8 CheckDynamicMoveType(struct Pokemon *mon, u16 move, u32 battler)
+{
+    u8 type = gMovesInfo[move].type;
+    u16 species = GetMonData(mon, MON_DATA_SPECIES);
+    u16 heldItem = GetMonData(mon, MON_DATA_HELD_ITEM, 0);
+    u16 heldItemEffect = ItemId_GetHoldEffect(heldItem);
+    u16 ability = GetMonAbility(mon);
+    u8 type1 = gSpeciesInfo[species].types[0];
+    u8 type2 = gSpeciesInfo[species].types[1];
+
+
+    if (!P_SHOW_DYNAMIC_TYPES)
+    {
+        return type;
+    }
```

Comment:
actually could you move this to where the function is invoked? In the attempt a user wants to add this feature they will see that it already exists.